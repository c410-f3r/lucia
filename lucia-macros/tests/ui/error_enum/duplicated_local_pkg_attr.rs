#![allow(incomplete_features)]
#![feature(async_fn_in_trait, impl_trait_projections)]

#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[pkg::aux]
  #[pkg::aux]
  impl Foo {}

  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req;

  #[pkg::res_data]
  struct Res;
}

#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[pkg::before_sending]
  #[pkg::before_sending]
  async fn before_sending() -> lucia::Result<()> {
    Ok(())
  }

  #[derive(Debug)]
  #[pkg::req_data]
  pub struct Req;

  #[pkg::res_data]
  struct Res;
}

#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[derive(Debug)]
  #[pkg::req_data]
  #[pkg::req_data]
  pub struct Req;

  #[pkg::res_data]
  struct Res;
}

fn main() {
}
