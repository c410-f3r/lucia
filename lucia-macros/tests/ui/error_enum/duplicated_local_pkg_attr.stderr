error: It is not possible to have more than one `pkg` attribute in the same element.
 --> tests/ui/error_enum/duplicated_local_pkg_attr.rs:6:3
  |
6 |   #[pkg::aux]
  |   ^^^^^^^^^^^

error: It is not possible to have more than one `pkg` attribute in the same element.
  --> tests/ui/error_enum/duplicated_local_pkg_attr.rs:20:3
   |
20 |   #[pkg::before_sending]
   |   ^^^^^^^^^^^^^^^^^^^^^^

error: It is not possible to have more than one `pkg` attribute in the same element.
  --> tests/ui/error_enum/duplicated_local_pkg_attr.rs:37:3
   |
37 |   #[pkg::req_data]
   |   ^^^^^^^^^^^^^^^^
